### PR TITLE
Adjust whitespace requirements on tag grammar rules (SYN-3737)

### PR DESCRIPTION
**Pull request by Cisphyx**
…mtagprop

### CODE REVIEW

File: synapse/lib/storm.lark [hunk 1]
```diff
@@ -180,18 +180,20 @@ edgedeln1: _EDGEN1INIT _valu _EDGEN1FINI baresubquery
...
-liftformtag: (PROPS | UNIVNAME) tagname [_cmpr _valu]
+liftformtag: (PROPS | UNIVNAME) tagnamenospace [_cmpr _valu]
 _DEREF.3:  /\*(?=\$)/
 liftpropby: ((PROPS | UNIVNAME) | _DEREF _varvalu) _cmpr _valu
 liftprop: ((PROPS | UNIVNAME) | _DEREF _varvalu)
 liftbyarray: ((PROPS | UNIVNAME) | _DEREF _varvalu) "*[" _safe_cmpr _valu "]"
-lifttagtag: "#" tagname [_cmpr _valu]
-liftbytag: tagname [_cmpr _valu]
+lifttagtag:(_HASHSPACE | _HASHNOSPACE) tagnamenospace [_cmpr _valu]
+liftbytag: (tagname | tagnamenospace) [_cmpr _valu]
```
Comment (invisig0th) [inline synapse/lib/storm.lark:189]: Would it be easier to make the lift-by-tag into a single token that we chop up in AST?

**Review by invisig0th — CHANGES_REQUESTED** [review]

File: synapse/lib/storm.lark [hunk 2]
```diff
@@ -221,7 +223,12 @@ cmdrargs: _cmdrargv (EQNOSPACE? _cmdrargv)*
 _cmdrargv: baresubquery | DOUBLEQUOTEDSTRING | SINGLEQUOTEDSTRING | CMDRTOKN | CMDOPT (EQNOSPACE (CMDRTOKN | SINGLEQUOTEDSTRING | DOUBLEQUOTEDSTRING | baresubquery))?
 
 // A tag with either a variable reference or a tag name potentially with asterisks
-_tagmatch: "#" (_varvalu | TAGMATCH)
+_tagmatch: _HASHSPACE (_varvalu | TAGMATCH)
+_HASHSPACE.2: /((?<=\s)|^)#/
```
[the remark this answers is not in the document]
Comment (Cisphyx) [reply synapse/lib/storm.lark:227]: This should only care about space/nospace, the ^ wasn't necessary.

File: synapse/lib/storm.lark (hunk 1 above)
Comment (Cisphyx) [reply synapse/lib/storm.lark:189]: This feels like it ends up with even more regexing and doesn't actually simplify it.

Comment (invisig0th) [reply synapse/lib/storm.lark:189]: Sorry, i meant lift by form tag.

**Review by invisig0th — APPROVED** [review]

**Review by invisig0th — APPROVED** [review]

**Pull request merged**